lib/plugins/aws/deploy/compile/functions/index.js

```diff
@@ -81,6 +81,19 @@ class AwsCompileFunctions {
       newFunction.Properties.Description = functionObject.description;
     }
 
+    newFunction.Properties.Environment.Variables = Object.assign(
```

Comment:
Is there a way to create this only if an `environment` property is given on the `function` or `provider` level?

This way we won't have to add the empty property to every test (and reduce the template size a little bit).